### ISSUE TITLE
isnan in complex functions triggers warning "calling a __host__ function ..."

### ISSUE THREAD

**Issue by schiller-manuel** [thrust/thrust#476]
When including _thrust/complex.h_ I get a lot of the following error messages:

```
/opt/thrust/thrust/detail/complex/catrig.h(613): warning: calling a __host__ function from a __host__ __device__ function is not allowed
```

Apparently, the usage of _isnan()_ triggers this warning. This might be a warning one can ignore, but maybe some #include is missing or wrong ...
